operator/controllers/conditions.go

```diff
@@ -0,0 +1,91 @@
+package controllers
+
+import (
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Reason = string
+
+const (
+	ReconcileSucceeded     Reason = "ReconcileSucceeded"
+	Initialized            Reason = "Initialized"
+	Processing             Reason = "Processing"
+	OlderCRExists          Reason = "OlderCRExists"
+	ChartInstallFailed     Reason = "ChartInstallFailed"
+	ConsistencyCheckFailed Reason = "ConsistencyCheckFailed"
+	MissingSecret          Reason = "MissingSecret"
+	InvalidSecret          Reason = "InvalidSecret"
+	HardDeleting           Reason = "HardDeleting"
+	ResourceRemovalFailed  Reason = "ResourceRemovalFailed"
+	HardDeleteFailed       Reason = "HardDeleteFailed"
+	SoftDeleteFailed       Reason = "SoftDeleteFailed"
+	SoftDeleting           Reason = "SoftDeleting"
+	Recovered              Reason = "Recovered"
+)
+
+type TypeAndStatus struct {
+	Status metav1.ConditionStatus
+	Type   string
+}
+
+var Ready = TypeAndStatus{
+	Status: metav1.ConditionTrue,
+	Type:   "Ready",
+}
+
+var NotReady = TypeAndStatus{
+	Status: metav1.ConditionFalse,
+	Type:   "Ready",
+}
+
+var Reasons = map[Reason]TypeAndStatus{
+	ReconcileSucceeded:     Ready,
+	Initialized:            Ready,
```

Comment:
I would consider `Initialized` Reason as beginning of the provisioning process, hence BtpOperator is not Ready to use yet at that time

Reply:
In the next commit.